File: Freestylecoding.Math.FSharp.Tests/Natural/UnaryPlusOperators.fs
namespace Natural

open Xunit
open Freestylecoding.Math

type public UnaryPlusOperators() =
    static member Plus( n:'T when 'T :> System.Numerics.IUnaryPlusOperators<'T,'T> ) : 'T =
        'T.op_UnaryPlus( n )

    [<Theory>]
    [<InlineData(  0u )>]   // Sanity
    [<InlineData(  1u )>]   // Sanity
    [<InlineData( 42u )>]   // Sanity
    member public this.Sanity value =
        Assert.Equal(
            Natural( [value] ),
            UnaryPlusOperators.Plus( Natural( [value] ) )
        )

File: Freestylecoding.Math.CSharp.Tests/Natural/UnaryPlusOperators.cs
using System.Numerics;
using Xunit;

namespace Freestylecoding.Math.CSharp.Tests.Natural;

using static Freestylecoding.Math.CSharp.Tests.Helpers;
using Natural = Freestylecoding.Math.Natural;

public class UnaryPlusOperators {
	private static T Plus<T>( T n ) where T : IUnaryPlusOperators<T, T> =>
		+n;
	[Theory]
	[InlineData( 0u )]   // Sanity
	[InlineData( 1u )]   // Sanity
	[InlineData( 42u )]   // Sanity
	public void Sanity( uint value ) =>
		Assert.Equal(
			new Natural( value ),
			Plus( new Natural( value ) )
		);
}